### PR TITLE
Optimized the submission details page.

### PR DESCRIPTION
**Pull request by deyordanov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1577

### CODE REVIEW

**Comment by deyordanov** [comment]
After a lot of optimizations, the `submission details` page kept on loading slowly every once in a while ( it took up to `5-10 seconds` for the page to load sometimes ). After a lot of testing, adding the following  `non-clustered`  database indexes, improves the performance ( for the `submission details page` ):
1. `IsDeleted` for the `Submissions` table.
2. `SubmissionId` + `TestId` for the `TestRuns` table ( composite index ).

These queries can be used for creating the following indexes:
1. For the `Submissions` table:
  ```sql
  CREATE NONCLUSTERED INDEX IX_TestRuns_Covering
  ON TestRuns (SubmissionId, TestId)
  ```
2. For the `TestRuns` table:
  ```sql
  CREATE NONCLUSTERED INDEX IX_TestRuns_SubmissionId_And_TestId
  ON TestRuns (SubmissionId, TestId)
  ```
Further improving the performance for the query related to the `TestRuns` table, can be done through a `non-clustered covering index`, but it has its `drawbacks` and should be discussed further:
  ```sql
  CREATE NONCLUSTERED INDEX IX_TestRuns_Covering
  ON TestRuns (SubmissionId, TestId)
  INCLUDE (Id, TimeUsed, MemoryUsed, ExecutionComment, CheckerComment, ResultType, ExpectedOutputFragment, UserOutputFragment, IsTrialTest);
  ```

File: Services/UI/OJS.Services.Ui.Models/Submissions/SubmissionDetailsServiceModel.cs [hunk 1]
```diff
@@ -60,42 +53,68 @@ public class SubmissionDetailsServiceModel : IMapExplicitly
...
+                .ForMember(d => d.ContestCategoryId, opt => opt.MapFrom(s => s.Participant.Contest.CategoryId))
                 .ForMember(d => d.MaxPoints, opt => opt.MapFrom(s => s.Problem.MaximumPoints))
                 .ForMember(d => d.TotalTests, opt => opt.Ignore())
                 .ForMember(s => s.UserIsInRoleForContest, opt => opt.Ignore())
-                .ForMember(s => s.IsEligibleForRetest, opt => opt.Ignore());
+                .ForMember(s => s.IsEligibleForRetest, opt => opt.Ignore())
+                .ForMember(s => s.Tests, opt => opt.Ignore())
+                .ForMember(s => s.TestRuns, opt => opt.Ignore());
+
+        private static (long? MaxMemoryUsed, int? MaxTimeUsed) GetMaxMemoryAndTimeUsed(string? testRunsCache)
```
Comment (gogo4ds) [inline Services/UI/OJS.Services.Ui.Models/Submissions/SubmissionDetailsServiceModel.cs:90]: Extract this into common method and reuse it in all models, where needed.

**Review by gogo4ds — CHANGES_REQUESTED** [review]

**Comment by gogo4ds** [comment]
Can you check the index creation in model builder configuration:
CREATE NONCLUSTERED INDEX IX_TestRuns_SubmissionId_And_TestId
ON TestRuns (SubmissionId, TestId)

File: Services/UI/OJS.Services.Ui.Models/Submissions/SubmissionDetailsServiceModel.cs (hunk 1 above)
Comment (deyordanov) [reply Services/UI/OJS.Services.Ui.Models/Submissions/SubmissionDetailsServiceModel.cs:90]: Done.

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**